File: sdk/js/wallet.ts
import { web3Enable, web3Accounts, type InjectedAccountWithMeta } from '@polkadot/extension-dapp';

export interface WalletConnection {
  chain: 'polkadot' | 'ethereum';
  accounts: string[] | InjectedAccountWithMeta[];
}

export async function connectWallet(chain: 'polkadot' | 'ethereum'): Promise<WalletConnection> {
  if (chain === 'polkadot') {
    const extensions = await web3Enable('PolkaConnect SDK');
    if (extensions.length === 0) {
      throw new Error('No Polkadot.js extension found. Please install it from polkadot.js.org/extension');
    }
    const accounts = await web3Accounts();
    if (accounts.length === 0) {
      throw new Error('No Polkadot accounts found. Please create one in your extension.');
    }
    return { chain, accounts };
  }

  if (chain === 'ethereum') {
    if (typeof window === 'undefined' || !window.ethereum) {
      throw new Error('MetaMask not found. Please install it from metamask.io');
    }
    
    const accounts = await window.ethereum.request({ 
      method: 'eth_requestAccounts' 
    }) as string[];
    
    return { chain, accounts };
  }

  throw new Error(`Unsupported chain: ${chain}`);
}

export interface BalanceResponse {
  balance: string;
  formatted: string;
  chain: string;
  status: string;
}

export async function getWalletBalance(
  chain: string, 
  address: string,
  baseUrl: string = 'https://polkaconnect.replit.app'
): Promise<BalanceResponse> {
  const endpoint = `${baseUrl}/api/balance?chain=${chain}&address=${address}`;
  const response = await fetch(endpoint);
  
  if (!response.ok) {
    throw new Error(`Failed to fetch balance: ${response.statusText}`);
  }
  
  return response.json();
}

File: sdk/python/polkaconnect/wallet.py
"""Wallet connection and balance queries"""

import requests
from typing import Dict, Any


def connect_wallet(chain: str) -> str:
    """
    Simulate wallet connection (for demonstration)

    In a real implementation, this would integrate with:
    - Polkadot.js extension via browser interface
    - MetaMask via web3.py or similar

    Args:
        chain: Chain name ('polkadot' or 'ethereum')

    Returns:
        Connection confirmation message

    Raises:
        ValueError: If unsupported chain specified
    """
    if chain not in ['polkadot', 'ethereum']:
        raise ValueError(f"Unsupported chain: {chain}")
    return f"Simulated connection to {chain} wallet"


def get_balance(
    chain: str, 
    address: str, 
    base_url: str = "https://polkaconnect.replit.app"
) -> Dict[str, Any]:
    """
    Get wallet balance for a specific chain and address

    Args:
        chain: Chain identifier (e.g., 'polkadot', 'moonbeam')
        address: Wallet address
        base_url: Base URL of the PolkaConnect API

    Returns:
        Balance information including formatted amount and status

    Raises:
        requests.HTTPError: If the API request fails
    """
    url = f"{base_url}/api/balance"
    params = {"chain": chain, "address": address}
    
    response = requests.get(url, params=params)
    response.raise_for_status()
    
    return response.json()
